zookeeper-server/src/test/java/org/apache/zookeeper/test/ClientSSLTest.java

```diff
@@ -156,4 +161,26 @@ public void testSecureStandaloneServer() throws Exception {
         mt.shutdown();
     }
 
+    @Test
+    public void testSecureStandaloneServerAuthFail() throws IOException {
+        try {
+            System.setProperty(ProviderRegistry.AUTHPROVIDER_PROPERTY_PREFIX + "authfail",
+                "org.apache.zookeeper.test.AuthFailX509AuthenticationProvider");
```

Comment:
nit:
you can use `AuthFailX509AuthenticationProvider.class.getName()`
this way this code won't be subject to problems in case of refactor/move class/rename class